File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/other_settings/OtherSettings.java
package com.github.alexandrecarlton.idea.settings.layout.other_settings;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.other_settings.checkstyle.CheckstyleSettings;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableOtherSettings.class)
public interface OtherSettings {
  Optional<CheckstyleSettings> checkstyle();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/other_settings/OtherSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.other_settings

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.other_settings.checkstyle.CheckstyleSettings
import com.github.alexandrecarlton.idea.settings.layout.other_settings.save_actions.SaveActionsSettings

data class OtherSettings(
    @JsonProperty("Checkstyle")
    val checkstyle: CheckstyleSettings? = null,

    @JsonProperty("Save Actions")
    val saveActions: SaveActionsSettings? = null
)
